src/BitmovinPlayer.tsx

```diff
@@ -51,42 +49,37 @@ export const BitmovinPlayer = forwardRef(function BitmovinPlayer(
   const latestPlayerRef = useRef<PlayerAPI | undefined>();
 
   // Initialize the player on mount.
-  useEffect(
-    () => {
-      const rootContainerElement = rootContainerElementRef.current;
+  useEffect(() => {
+    const rootContainerElement = rootContainerElementRef.current;
 
-      if (!rootContainerElement) {
-        return;
-      }
+    if (!rootContainerElement) {
+      return;
+    }
 
-      // We create elements manually to workaround the React strict mode.
-      // In the strict mode the mount hook is invoked twice. Since the destroy method is async
-      // the next mount hook is invoked before the previous destroy method is finished and the new player instance
-      // messes up the old one. This workaround ensures that each player instance has its own container and video elements.
-      // This should be improved in the future if possible.
-      const { createdPlayerContainerElement, createdVideoElement } = preparePlayerElements(rootContainerElement);
+    // We create elements manually to workaround the React strict mode.
+    // In the strict mode the mount hook is invoked twice. Since the destroy method is async
+    // the next mount hook is invoked before the previous destroy method is finished and the new player instance
+    // messes up the old one. This workaround ensures that each player instance has its own container and video elements.
+    // This should be improved in the future if possible.
+    const { createdPlayerContainerElement, createdVideoElement } = preparePlayerElements(rootContainerElement);
 
-      const convertedConfig = convertConfig(config);
-      const initializedPlayer = initializePlayer(createdPlayerContainerElement, createdVideoElement, convertedConfig);
+    const convertedConfig = convertConfig(config);
+    const initializedPlayer = initializePlayer(createdPlayerContainerElement, createdVideoElement, convertedConfig);
 
-      initializePlayerUi(initializedPlayer, config, customUi);
+    initializePlayerUi(initializedPlayer, config, customUi);
 
-      latestPlayerRef.current = initializedPlayer;
+    latestPlayerRef.current = initializedPlayer;
 
-      if (playerRefProp) {
-        setRef(playerRefProp, initializedPlayer);
-      }
+    if (playerRefProp) {
+      setRef(playerRefProp, initializedPlayer);
+    }
 
-      setPlayer(initializedPlayer);
+    setPlayer(initializedPlayer);
 
-      return () => {
-        destroyPlayer(initializedPlayer, rootContainerElement, createdPlayerContainerElement);
-      };
-    },
-    // Ignore the dependencies, as the effect should run only once (on mount).
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [],
-  );
+    return () => {
+      destroyPlayer(initializedPlayer, rootContainerElement, createdPlayerContainerElement);
+    };
+  }, [config, customUi, playerRefProp]);
```

Comment:
As discussed, let's check if there are any bad effects from having the ref as a dependency.

Reply:
I think it can be done as a separate use effect hook to avoid player reinitialization on the ref property changes - https://github.com/bitmovin/bitmovin-player-react/pull/12/commits/f0703e60f851949528138dd07e3142ae9660c1d9.